kong/plugins/moesif/moesif_gov.lua

```diff
@@ -0,0 +1,310 @@
+local _M = {}
+local governance_rules_hashes = {}
+local ngx_timer_at = ngx.timer.at
+local ngx_log = ngx.log
+local ngx_log_ERR = ngx.ERR
+local ngx_md5 = ngx.md5
+local connect = require "kong.plugins.moesif.connection"
+local helper = require "kong.plugins.moesif.helpers"
+local log = require "kong.plugins.moesif.log"
+local string_format = string.format
+local req_get_headers = ngx.req.get_headers
+local cjson = require "cjson"
+local governance_rules_etags = {}
+local fetch_governance_rules = {}
+local socket = require "socket"
+
+-- Get Governance Rules function
+-- @param `premature`
+-- @param hash_key   Hash key of the config application Id
+-- @param `conf`     Configuration table, holds http endpoint details
+function get_governance_rules(premature, hash_key, conf)
+    if premature then
+    return
+    end
+
+    local sock, parsed_url = connect.get_connection("/v1/rules", conf)
+
+    -- Prepare the payload
+    local payload = string_format("%s %s HTTP/1.1\r\nHost: %s\r\nConnection: Keep-Alive\r\nX-Moesif-Application-Id: %s\r\n",
+                                  "GET", parsed_url.path, parsed_url.host, conf.application_id)
+
+    -- Send the request
+    local ok, err = sock:send(payload .. "\r\n")
+    if not ok then
+        -- Set fetch_governance_rule to true, to be able to fetch governance rule next time
+        fetch_governance_rules[hash_key] = true
```

Comment:
fetch_governance_rules[hash_key] = true  is set at the end of the function and I dont see any return statement before.. So no need to set this here and line 52?